src/services/userService.js

```diff
@@ -39,11 +39,32 @@ async function login(username, password) {
     }
 }
 
-async function getUserByUsername(username) {
-    const result = await userDAO.queryByUsername(username);
+const updateRole = async (id, role) => {
+    let result = await userDAO.getUserById(id);
+    throwIfError(result);
+    const foundUser = result.Item;
+    if (!foundUser) {
+        throw {
+            status: 400,
+            message: `User with id ${id} not found`
+        }
+    }
+
+    const currentRole = foundUser.role;
+    if (currentRole === role) {
+        throw {
+            status: 400,
+            message: `User is already role ${role}`
+        }
+    } else if (role !== "admin") {
+        throw {
+            status: 400,
+            message: "Cannot demote admin, use AWS console instead"
+        }
+    }
+
+    result = await userDAO.updateRole(id, role);
     throwIfError(result);
```

Comment:
Why not return result at the end of the function?

Reply:
I didn't have a particular use for the returned result, but I can add it in for a potential future use case.